packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -730,6 +732,20 @@ export class BasicCrawler<
                     requestQueue: await this.getRequestQueue(),
                 });
             },
+            sendRequest: async (options?: OptionsInit) => {
```

Comment:
This is quite ugly in the terms of types, because we don't know whether it's a Promise or a stream. So people will need to do:

```ts
const response = await crawlingContext.sendRequest() as Response<string>;
```

Reply:
I feel like we should drop the `isStream` option in Got and support only `got(...)` and `got.stream(...)`.

Reply:
If we have the `isStream` toggle inside `OptionsInit`, we can easily alter the return type based on that. I can help with that if you don't know how - ideally prepare some examples/tests where you use casting so I have a ground for testing.

Now seeing you have the `options` defined as optional, that's obviously not gonna work for such case, but I would expect users to pass the requests explicitly. And more importantly - the idea behind this was not to have a shotcut for `gotScraping` to fire the current request object, **but to allow processing any `Request` objects** (makes sense to process the bound request by default, but if provided, the request from arguments needs to be used on its own, not as an extension (= those should not be merged to a single options object). The `Request` interface is not compatible with `OptionsInit` interface, right? That was the whole point for this helper method.

> I feel like we should drop the isStream option in Got and support only got(...) and got.stream(...).

Yeah that makes a lot of sense to me, but let's focus on what we can do now.

Reply:
I see now. Will redo.